snuba/web/db_query.py

```diff
@@ -157,54 +156,33 @@ def raw_query(
-                            with_totals=request.query.has_totals(),
-                        )
+                    result = reader.execute(
+                        query,
+                        query_settings,
+                        # All queries should already be deduplicated at this point
+                        # But the query_id will let us know if they aren't
+                        query_id=query_id if use_deduper else None,
+                        with_totals=request.query.has_totals(),
+                    )
 
-                        timer.mark("execute")
-                        stats.update(
-                            {
-                                "result_rows": len(result["data"]),
-                                "result_cols": len(result["meta"]),
-                            }
-                        )
+                    timer.mark("execute")
+                    stats.update(
+                        {
+                            "result_rows": len(result["data"]),
+                            "result_cols": len(result["meta"]),
+                        }
+                    )
 
-                        if use_cache:
-                            cache.set(query_id, result)
-                            timer.mark("cache_set")
-
-                    except BaseException as ex:
-                        error = str(ex)
-                        logger.exception("Error running query: %s\n%s", sql, error)
-                        stats = update_with_status("error")
-                        meta = {}
-                        if isinstance(ex, ClickhouseError):
-                            err_type = "clickhouse"
-                            meta["code"] = ex.code
-                        else:
-                            err_type = "unknown"
-                        raise RawQueryException(
-                            err_type=err_type,
-                            message=error,
-                            stats=stats,
-                            sql=sql,
-                            **meta,
-                        )
-            except RateLimitExceeded as ex:
-                stats = update_with_status("rate-limited")
-                raise RawQueryException(
-                    err_type="rate-limited",
-                    message="rate limit exceeded",
-                    stats=stats,
-                    sql=sql,
-                    detail=str(ex),
-                )
+                    if use_cache:
+                        cache.set(query_id, result)
+                        timer.mark("cache_set")
+            except Exception as cause:
+                if isinstance(cause, RateLimitExceeded):
+                    stats = update_with_status("rate-limited")
+                else:
+                    logger.exception("Error running query: %s\n%s", sql, cause)
+                    stats = update_with_status("error")
+                raise RawQueryException(stats=stats, sql=sql) from cause
```

Comment:
If `stats` and `sql` are removed from the response body, `RawQueryException` can be removed and this exception can just be rethrown and handled in `dataset_query` without all of the unwrapping. That is my ideal scenario. Right now this will still yield strange stacktraces to anything that is not web.